game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/AaCasualtySelectorTest.java

```diff
@@ -0,0 +1,432 @@
+      assertThat(details.getDamaged(), is(empty()));
+      verify(bridge, description("10 planes with only 5 hits"))
+          .getRandom(eq(10), eq(5), any(), any(), anyString());
+    }
+
+    @Test
+    void hitsEqualToPlanesMultiHpDamagesAndKillsAll() {
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              givenMultiHpPlaneUnits(1),
+              givenDamageableAaUnits(1),
+              mock(CombatValue.class),
+              mock(CombatValue.class),
+              "text",
+              givenDiceRoll(true, true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat("Plane was killed", details.getKilled(), hasSize(1));
+      assertThat("Plane was also damaged", details.getDamaged(), hasSize(1));
+    }
+
+    @Test
+    void hitsGreaterThanPlanesMultiHpDamagesAndKillsAll() {
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              givenMultiHpPlaneUnits(1),
+              givenDamageableAaUnits(1),
+              mock(CombatValue.class),
+              mock(CombatValue.class),
+              "text",
+              givenDiceRoll(true, true, true, true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat("Plane was killed", details.getKilled(), hasSize(1));
+      assertThat("Plane was also damaged", details.getDamaged(), hasSize(1));
+    }
+
+    @Test
+    void oneHitAgainstMultiHpPlaneOnlyDamagesIt() {
+
+      whenGetRandom(bridge).thenAnswer(withValues(0));
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              givenMultiHpPlaneUnits(1),
+              givenDamageableAaUnits(1),
+              mock(CombatValue.class),
+              mock(CombatValue.class),
+              "text",
+              givenDiceRoll(true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat("Plane is not killed", details.getKilled(), is(empty()));
+      assertThat("Plane is damaged", details.getDamaged(), hasSize(1));
+    }
+
+    @Test
+    void threeHitsAgainstTwoMultiHpPlanesKillsOneAndDamagesTheOther() {
+
+      whenGetRandom(bridge).thenAnswer(withValues(0, 1, 2));
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              givenMultiHpPlaneUnits(2),
+              givenDamageableAaUnits(1),
+              mock(CombatValue.class),
+              mock(CombatValue.class),
+              "text",
+              givenDiceRoll(true, true, true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat("1 Plane is killed", details.getKilled(), hasSize(1));
```

Comment:
Why is 1 plane killed? The assertion here is not really checking that it's a plane that is killed either.

I ask that question as that is the question a maintainer would have after reading "1 plane is killed". IMO it paraphrases the asserted state rather than describing why we expect to be in that state. Perhaps if we said something like "Rolling a '0' on the random bridge is a hit"

That could probably be made even more clear as well by assigning the dice rolls to variables, eg:
```
int aaHit = 0;
int aaMiss = 2

whenGetRandom(bridge).thenAnswer(withValues(aaHit, aaMiss, aaMiss));
```


```

Reply:
The method name explains why exactly one of the planes are hit.  But I can move that to this message as well.

As for the dice rolls, they aren't related to hits.  They are used to randomly select the units that get the hits.

Reply:
An assert message and the test method should not be redundant to each other. If so, then the assert message is not needed. Generally the method name is what you are testing, what aspect of a feature, the assert method message is why the condition is expected to be true. Often when there is one assert in a method the reasoning is clear. Again, the goal is to help other developers know why the test is supposed to pass (not to check a box or to pass a style guide, but if someone is developing, breaks this test, they need to know why the test was correct to begin with and why the assertions were expected to pass)